### PR TITLE
feat: update partial tag names to follow Django convention

### PR DESCRIPTION
**Pull request by scriptogre**
- Rename 'startpartial' to 'partialdef'
- Update corresponding end tag to 'endpartialdef'
- Alias old 'startpartial' and add a DeprecationWarning when used
- Update existing test_partial_tags test case
- Add test_deprecated_startpartial_tag test case

BREAKING CHANGE: The tag name 'startpartial' is deprecated and will be removed in future versions. Use 'partialdef' instead.

Fixes issue #12 - templatetag incompatible with IntelliJ editors.

### CODE REVIEW

**Comment by carltongibson** [comment]
@scriptogre Fancy looking at the test errors here? Thanks.

File: src/template_partials/templatetags/partials.py [hunk 1]
```diff
@@ -56,33 +58,43 @@ def render(self, context):
...
 
-        {% endpartial %}
+        {% endpartialdef %}
 
     Stores the nodelist in the context under the key "partial_contents" and can
     be retrieved using the {% partial %} tag.
     """
-    # Parse the tag
+    _define_partial(parser, token, "endpartialdef")
+    return DefinePartialNode
```
Comment (carltongibson) [inline src/template_partials/templatetags/partials.py:78]: Surely... 

```suggestion
    return _define_partial(parser, token, "endpartialdef")
```

... no? 🤔

Comment (scriptogre) [reply src/template_partials/templatetags/partials.py]: Yup, don't know how I made that mistake :) 

I fixed the merge conflicts as well (with the new inline option).  Hope it's all good and merge-able now!

Comment (carltongibson) [reply src/template_partials/templatetags/partials.py]: Thanks for the update @scriptogre — I will take another look.

**Comment by carltongibson** [comment]
Merged into `main` as 4c3c9cdb0984670beab032162856f8cf73fcdde2. Thanks for this @scriptogre! It'll be in the next release shortly. 🎁

**Pull request closed without merging**